src/main/resources/io/jenkins/plugins/folderauth/FolderAuthorizationStrategyManagementLink/index.jelly

```diff
@@ -110,15 +116,18 @@
                                     <br/>
                                     ${%folders}: ${folderRole.getFolderNamesCommaSeparated()}
                                     <br/>
-                                    <form action="${rootURL}/${it.urlName}/assignSidToFolderRole" method="POST">
+                                    <form action="${rootURL}/${it.urlName}/assignOrRemoveSidFolderRole" method="POST">
                                         <div class="form-row">
-                                            <label for="assign-sid-folder-${index}" style="margin-right: 10px;">
-                                                ${%assignSid}
+                                            <label for="assign-sid-folder-${folderRole.name}" style="margin-right: 10px;">
+                                                SID
                                             </label>
-                                            <input id="assign-sid-folder-${index}" type="text" name="sid"/>
+                                            <input id="assign-sid-folder-input-${folderRole.name}" type="text" name="sid"/>
                                             <input type="hidden" value="${folderRole.name}" name="roleName"/>
-                                            <input type="submit" style="margin-left: 10px;" value="${%submit}"
-                                                   class="submit"/>
+                                            <br/>
+                                            <input id="assign-sid-folder-submit-${folderRole.name}" type="submit"
+                                                   style="margin-left: 40px;" value="Assign" name="action" class="submit"/>
```

Comment:
This works but I was originally thinking about an XHR from JavaScript. @oleg-nenashev What would you suggest here? Continue as is or move the logic to JavaScript and avoid adding the `doAssignRemoveSidGlobalRole` and similar API methods?

Another idea: `doAssignRemoveSid` with the role type as a query parameter.

Reply:
yea having a `doAssignRemoveSid` with role type as parameter would be best, less repeated code. 

hmm...do you mind accepting this first though? I don't mind refactoring it but I'm trying to get this plugin up and running first to replace the role strategy plugin as I have a page on my production server that's loading very slowly.